docs/engineering/Testing/ui-tests.md

```diff
@@ -0,0 +1,550 @@
+---
+id: ui-tests
+title: UI Tests with Jest and Storybook
+---
+
+This guide will cover how to write tests for the bobaboard-ui codebase. It assumes that you have some familiarity with Bobaboard as a user of the site, have taken at least a brief look at the file structure of the [bobaboard-ui codebase](https://github.com/essential-randomness/bobaboard-ui), and have a basic understanding of javascript, [Typescript](https://www.typescriptlang.org/docs/handbook/typescript-from-scratch.html), and React components and props. If you would like to brush up on React, the React docs offer a number of great [resources for learning](https://reactjs.org/docs/getting-started.html#learn-react).
```

Comment:
I understand why this is here, but I think we need a more concise way to express this. Maybe we can do a custom callout that says:

```
::: prerequisites
- Basics of testing (link to doc)
- Jest (I'd write this as a separate guide since we use it on the backend too, and that has no storybook)
- React (I would say typescript and JS are a prerequisite of React). 
:::
```

We can have an intro page for React with resources, and link that here. You can also link the last bulletpoint to the react guide for now.

If you want to see how to create a callout, refer to this: https://github.com/essential-randomness/bobadocs/blob/main/docusaurus.config.js#L146

Reply:
Note that if we write Jest as a separate guide, there's still Jest assertions that only make sense in this context (e.g. `toBeVisible`, so we can keep those for this guide and simply mention it from the Jest guide.